cobalt/browser/h5vcc_system/h5vcc_system_impl_tvos.mm

```diff
@@ -0,0 +1,149 @@
+  std::move(callback).Run(base::SysNSStringToUTF8(advertising_id));
+}
+
+bool GetLimitAdTrackingShared() {
+  return [ATTrackingManager trackingAuthorizationStatus] !=
+         ATTrackingManagerAuthorizationStatusAuthorized;
+}
+
+std::string GetTrackingAuthorizationStatusShared() {
+  ATTrackingManagerAuthorizationStatus status =
+      [ATTrackingManager trackingAuthorizationStatus];
+  switch (status) {
+    case ATTrackingManagerAuthorizationStatusAuthorized:
+      return std::string("AUTHORIZED");
+    case ATTrackingManagerAuthorizationStatusDenied:
+      return std::string("DENIED");
+    case ATTrackingManagerAuthorizationStatusRestricted:
+      return std::string("RESTRICTED");
+    case ATTrackingManagerAuthorizationStatusNotDetermined:
+      return std::string("NOT_DETERMINED");
+  }
+  NOTREACHED();
+}
+
+void PerformExitStrategy() {
+  SbSystemRequestConceal();
+}
+
+}  // namespace
+
+void H5vccSystemImpl::GetAdvertisingId(GetAdvertisingIdCallback callback) {
+  DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+  GetAdvertisingIdShared(std::move(callback));
+}
+
+void H5vccSystemImpl::GetAdvertisingIdSync(
+    GetAdvertisingIdSyncCallback callback) {
+  DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+  GetAdvertisingIdShared(std::move(callback));
+}
+
+void H5vccSystemImpl::GetLimitAdTracking(GetLimitAdTrackingCallback callback) {
+  DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+  std::move(callback).Run(GetLimitAdTrackingShared());
+}
+
+void H5vccSystemImpl::GetLimitAdTrackingSync(
+    GetLimitAdTrackingSyncCallback callback) {
+  DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+  std::move(callback).Run(GetLimitAdTrackingShared());
+}
+
+void H5vccSystemImpl::GetTrackingAuthorizationStatus(
+    GetTrackingAuthorizationStatusCallback callback) {
+  DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+  std::move(callback).Run(GetTrackingAuthorizationStatusShared());
+}
+
+void H5vccSystemImpl::GetTrackingAuthorizationStatusSync(
+    GetTrackingAuthorizationStatusSyncCallback callback) {
+  DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+  std::move(callback).Run(GetTrackingAuthorizationStatusShared());
+}
+
+void H5vccSystemImpl::RequestTrackingAuthorization(
+    RequestTrackingAuthorizationCallback callback) {
+  DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+  auto completion = base::CallbackToBlock(base::BindPostTask(
+      base::SequencedTaskRunner::GetCurrentDefault(),
+      base::BindOnce(
+          [](RequestTrackingAuthorizationCallback callback,
+             ATTrackingManagerAuthorizationStatus /*status*/) {
+            std::move(callback).Run();
+          },
+          std::move(callback))));
+  [ATTrackingManager
+      requestTrackingAuthorizationWithCompletionHandler:completion];
+}
+
+void H5vccSystemImpl::GetUserOnExitStrategy(
+    GetUserOnExitStrategyCallback callback) {
+  std::move(callback).Run(h5vcc_system::mojom::UserOnExitStrategy::kMinimize);
+}
+
+void H5vccSystemImpl::Exit() {
```

Comment:
The method of this body is the same for the other platforms, 
could we refactor it?

Reply:
Hmm true, this patch was written before all these cookie-related changes from #8130. I think it makes sense to move this to `_base.cc` and have each platform file implement `PerformExitStrategy()` instead.